lib/validate.js

```diff
@@ -1,10 +1,29 @@
 var ajv = require('ajv')();
-var schema = require('./schema.json');
+var sb2schema = require('./sb2_schema.json');
+var sb3schema = require('./sb3_schema.json');
 
 module.exports = function (input, callback) {
-    var validate = ajv.compile(schema);
-    var valid = validate(input);
+    var validateSb2 = ajv.compile(sb2schema);
+    var validateSb3 = ajv.compile(sb3schema);
 
-    if (!valid) return callback(validate.errors);
-    callback(null, input);
+    var isValidSb2 = validateSb2(input);
+
+    if (isValidSb2) {
+        input.projectVersion = 2;
+        return callback(null, input);
+    }
+
+    var isValidSb3 = validateSb3(input);
+    if (isValidSb3) {
+        input.projectVersion = 3;
+        return callback(null, input);
+    }
+
+    var validationErrors = {
+        validationError: 'Could not parse as a valid SB2 or SB3 project.',
+        sb2Errors: validateSb2.errors,
+        sb3Errors: validateSb3.errors
+    };
```

Comment:
In the long run it may end up cleaner and/or faster to offer just one error from this list:
- I couldn't determine the project format.
- I determined that this project is in format `X` but I ran into error `Y` parsing it as such.

I think it would make sense to include some tag for this purpose - a format version number, maybe - in the SB3 spec.

Reply:
Agreed; I think [`meta.semver`](https://github.com/kchadha/scratch-parser/blob/03ca32717de35370cc497d3f7d1e8b43f535d059/lib/sb3_schema.json#L329-L332) could work for this use?

Reply:
Hmm @cwillisf, I think it's better to give the users of this library as much information as possible about why the parse failed. Especially since 'analyze' will be moving out of here, the sole purpose of this library will be to attempt to parse arbitrary things as sb2 or sb3 and tell users whether one of them passed or why they both failed.
If there are multiple errors, the user has hope of attempting to resolve all of them before running the parser again instead of continuously re-parsing after every error resolution.

I also don't want to use some arbitrary property to 'decide' that it's one format or the other and then only attempt to parse it as that one format. I could easily add meta.semver = 3.0.0 to an sb2's project.json, but that doesn't make it an sb2. The entire point of this repository is for a user to plug in something they think is a scratch project and for the parser to tell them whether it is, and if so, what kind.

Reply:
Ah, OK. I agree that this pattern makes sense for that sort of use case, then. I was thinking about communicating to the user when an error happens during a regular load / import, though, where we probably just want one message.